Tribler/Core/Modules/restapi/downloads_endpoint.py

```diff
@@ -208,3 +209,37 @@ def render_PATCH(self, request):
                 return json.dumps({"error": "unknown state parameter"})
 
         return json.dumps({"modified": True})
+
+
+class DownloadExportTorrentEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for requests that are exporting a download to a .torrent file.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+        self.infohash = infohash
+
+    def render_GET(self, request):
+        """
+        .. http:get:: /download/(string: infohash)/torrent
+
+        A GET request to this endpoint returns the .torrent file associated with the specified download.
+
+            **Example request**:
+
+                .. sourcecode:: none
+
+                    curl -X GET http://localhost:8085/download/4344503b7e797ebf31582327a5baae35b11bda01/torrent
+
+            **Example response**:
+
+            The contents of the .torrent file.
+        """
+        download = self.session.get_download(self.infohash)
+        if not download:
+            return DownloadExportTorrentEndpoint.return_404(request)
+
+        request.setHeader(b'content-type', 'application/x-bittorrent')
```

Comment:
It's necessary to make web browsers happy (so they download the file immediately) :)